src/FSharp.Data.GraphQL.Server/Executor.fs

```diff
@@ -59,6 +65,44 @@ type Executor<'Root> (schema: ISchema<'Root>) =
                 let msg = ex.ToString()
                 return prepareOutput(Direct(new Dictionary<string, obj>() :> Output, [msg, []]))
         }
+
+    let execute (executionPlan: ExecutionPlan, data: 'Root option, variables: Map<string, obj> option) = 
+        eval(executionPlan, data, defaultArg variables Map.empty)
+
+    let rec runMiddlewares executionPlan data variables (middlewares : ExecutionMiddlewareFunc<'Root> list) =
+        match middlewares with
+        | [] -> execute (executionPlan, data, variables)
+        | [x] -> x (executionPlan, data, variables) (fun (plan, data, variables) -> runMiddlewares plan data variables [])
+        | x :: xs -> x (executionPlan, data, variables) (fun (plan, data, variables) -> runMiddlewares plan data variables xs)
+
+    let executeWithMiddlewares (executionPlan: ExecutionPlan, data: 'Root option, variables: Map<string, obj> option) =
+        middlewares
+        |> Seq.map (fun m -> m.AsyncResolve)
+        |> List.ofSeq
+        |> runMiddlewares executionPlan data variables
+
+    new(schema) = Executor(schema, middlewares = Seq.empty)
+
+    new(schema, middlewareFuncs : ExecutionMiddlewareFunc<'Root> seq) = 
```

Comment:
If the user wants to use plain functions as middleware, they can through the use of this constructor. Works exactly as the interface option.